chain/default_syncer.go

```diff
@@ -246,6 +254,12 @@ func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.Tip
 		return nil
 	}
 
+	for _, b := range next.ToSlice() {
+		if err := syncer.consensus.ValidateSemantic(ctx, parent.At(0), b); err != nil {
```

Comment:
I think @ZenGround0 was suggesting this check actually move into RunStateTransition, so this call becomes unnecessary. I have a slight preference for that too. When the syntax check moves out to the boundary, the net will be zero changes to syncer.